graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java

```diff
@@ -155,21 +155,22 @@ protected abstract HttpResponse executeGraphql(ServiceRequestContext ctx, Graphq
         return parseJsonString(value);
     }
 
-    private static Map<String, Object> toMap(@Nullable Object maybeMap) {
+    /**
+     * This only works reliably if maybeMap is from Json, as maps(objects) in Json
+     * can only have string keys.
+     */
+    @SuppressWarnings("unchecked")
+    private static Map<String, Object> toMapFromJson(@Nullable Object maybeMap) {
         if (maybeMap == null) {
             return ImmutableMap.of();
         }
 
         if (maybeMap instanceof Map) {
-            final Map<?, ?> variablesMap = (Map<?, ?>) maybeMap;
-            if (variablesMap.isEmpty()) {
+            final Map<?, ?> map = (Map<?, ?>) maybeMap;
+            if (map.isEmpty()) {
                 return ImmutableMap.of();
             }
-
-            final ImmutableMap.Builder<String, Object> builder =
-                    ImmutableMap.builderWithExpectedSize(variablesMap.size());
-            variablesMap.forEach((k, v) -> builder.put(String.valueOf(k), v));
-            return builder.build();
+            return (Map<String, Object>) map;
```

Comment:
Q) Is there a reason we're not wrapping in `Collections.unmodifiableMap()`?

```suggestion
            return (Map<String, Object>) Collections.unmodifiableMap(map);
```

Reply:
I thought it is fine because `parseJsonString()` in `toMap()` already returns a mutable `Map`.
If we decided to pass an immutable `Map` to `GraphqlRequest`, it would be nice to apply the `Map` returned by `parseJsonString()` too.

Reply:
I see. Sounds like this might be out of scope for this PR then 👍

Reply:
Even before this PR, the `Map` may (and often will) have other maps as values, and they were never converted into immutable maps. That's actually how we're currently circumventing the bug in Armeria, ie. by having complex types.

Imho I even think it's a bad idea to do the immutability, as Armeria is never using the map (just deserializing it and then sending it on), and forwarding immutable maps to Graphql-java when afaik there is no contract stating that immutable maps are ok, could very well break future versions of Graphql-java for no good reason.